python/src/iceberg/exceptions.py

```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class OperationError(Exception):
```

Comment:
Can you add some docs to this class? What is the purpose of `OperationError`?

Reply:
Is there a python equivalent of `IllegalArgumentException`? It would be nice to not need custom exceptions in so many cases. If not, then we should probably create `IcebergException` as a base class and create our own more general exception for cases like this rather than creating an exception specific to a class.

Reply:
Good points, an exception type for every class is definitely too much. There's the builtin `ValueError` that I can use here and then later as requirements for custom exceptions come up, I'll use your idea of creating a base `IcebergException` and building custom exceptions on top of that.

Reply:
fixed